src/Avalonia.Controls/IApplicationLifecycle.cs

```diff
@@ -7,14 +7,25 @@ namespace Avalonia.Controls
     /// </summary>
     public interface IApplicationLifecycle
     {
+        /// <summary>
+        /// Sent when the application is starting up.
+        /// </summary>
+        event EventHandler<StartupEventArgs> Startup;
+
         /// <summary>
         /// Sent when the application is exiting.
         /// </summary>
-        event EventHandler OnExit;
+        event EventHandler<ExitEventArgs> Exit;
+
+        /// <summary>
+        /// Shuts down an application that returns the specified exit code to the operating system.
+        /// </summary>
+        void Shutdown();
 
         /// <summary>
-        /// Exits the application.
+        /// Shuts down an application that returns the specified exit code to the operating system.
```

Comment:
I think a better wording might be:

```suggestion
        /// Shuts down the application, returning the specified exit code to the operating system.
```